src/core/SIP/Channels/SIPUDPChannel.cs

```diff
@@ -78,58 +64,43 @@ private void Dispose(bool disposing)
             }
         }
 
-        private void Listen()
+        private async Task Listen()
         {
-            try
-            {
-                byte[] buffer = null;
-
-                logger.LogDebug("SIPUDPChannel socket on " + m_localSIPEndPoint.ToString() + " listening started.");
+            logger.LogDebug("SIPUDPChannel socket on " + m_localSIPEndPoint.ToString() + " listening started.");
 
-                while (!Closed)
+            while (!Closed)
+            {
+                try
                 {
-                    IPEndPoint inEndPoint = new IPEndPoint(IPAddress.Any, 0);
-
-                    try
-                    {
-                        buffer = m_sipConn.Receive(ref inEndPoint);
-                    }
-                    catch (SocketException)
-                    {
-                        // ToDo. Pretty sure these exceptions get thrown when an ICMP message comes back indicating there is no listening
-                        // socket on the other end. It would be nice to be able to relate that back to the socket that the data was sent to
-                        // so that we know to stop sending.
-                        //logger.LogWarning("SocketException SIPUDPChannel Receive (" + sockExcp.ErrorCode + "). " + sockExcp.Message);
-
-                        //inEndPoint = new SIPEndPoint(new IPEndPoint(IPAddress.Any, 0));
-                        continue;
-                    }
-                    catch (Exception listenExcp)
-                    {
-                        logger.LogError("Exception listening on SIPUDPChannel. " + listenExcp.Message);
-                        inEndPoint = new IPEndPoint(IPAddress.Any, 0);
-                        continue;
-                    }
-
-                    if (buffer == null || buffer.Length == 0)
-                    {
-                        // No need to care about zero byte packets.
-                        //string remoteEndPoint = (inEndPoint != null) ? inEndPoint.ToString() : "could not determine";
-                        //logger.LogError("Zero bytes received on SIPUDPChannel " + m_localSIPEndPoint.ToString() + ".");
-                    }
-                    else
+                    var receiveResult = await m_sipConn.ReceiveAsync();
+                    if (receiveResult.Buffer != null && receiveResult.Buffer.Length > 0)

```

Comment:
Could just use `receiveResult.Buffer?.Length > 0`.

Reply:
Yeap. Fixed this as well.